app/controllers/samples_controller.rb

```diff
@@ -318,16 +332,34 @@ def find_index_assets
     end
   end
 
-  def find_samples(samples, link, options, title)
+  # Filters linked samples based on the provided options and template attribute title.
+  #
+  # @param samples [Array<Sample>] the list of samples to filter
+  # @param link [Symbol] the method to call on each sample to get the linked samples (:linked_samples or :linking_samples)
+  # @param options [Hash] the options for filtering
+  # @option options [Integer] :template_id the ID of the template to filter by
+  # @option options [String] :attribute_value the value of the attribute to filter by
+  # @option options [Integer] :attribute_id the ID of the attribute to filter by
+  # @param template_attribute_title [String] the title of the template attribute to filter by
+  # @return [Array<Sample>] the filtered list of samples
+  def filter_linked_samples(samples, link, options, template_attribute)
+    template_attribute_title = template_attribute&.title
     samples.select do |s|
       s.send(link).any? do |x|
```

Comment:
i think this is potentially dangerous, maybe not now but could be due to an error in the future. Just needs to check the `link` is in a list of allowed method calls, or have another way of specifying which method is called.

Reply:
Indeed! makes sense.
Now it will raise an error if `link` is not either `:linked_samples` or `:linking_samples`.